### ISSUE TITLE
1.2.1 regression vs 1.1.0

### ISSUE THREAD

**Issue by diegomontoya** [bluelinelabs/LoganSquare#100]
@EricKuck 

File: MyPacket$$JsonObjectMapper.java

First few lines generated by 1.1.0

```java
@SuppressWarnings("unsafe")
public final class MyPacket$$JsonObjectMapper extends JsonMapper<MyPacket> {
  @Override
  public MyPacket parse(JsonParser jsonParser) throws IOException {
    return _parse(jsonParser);
  }
```

First few lines generated by 1.2.1 which doesn't compile

```java
@SuppressWarnings("unsafe,unchecked")
public final class MyPacket$$JsonObjectMapper extends JsonMapper<MyPacket> {
  private static final JsonMapper<MyPacket> parentObjectMapper = JsonMapperLoaderImpl.COM_TEST_PACKET_COM_TEST_PACKET_MYPACKET;
```

Compile error is that it can't find 

```java
JsonMapperLoaderImpl.COM_TEST_PACKET_COM_TEST_PACKET_MYPACKET
```

Original Object:

```java

@JsonObject
public class MyPacket {
    @JsonField
    public BaseItem[] Items;

    @JsonField
    public int Var1;
    @JsonField
    public int Var2;
}
```

Item is also an JsonObject object with standard string/int/boolean fields. 

Everything works in 1.1.0 but not in 1.2.1 where generated java mapper file for this JsonObject can't compile.

**Comment by EricKuck** [comment]
Is there a way you could put together a quick sample project I could run to test this out? When I compile the exact code you sent, it works great for me. I'm also not getting the `parentObjectMapper` field the way you are.

**Comment by EricKuck** [comment]
Can you clean your project and try this again in 1.2.2 to see if that fixes it? If not, a sample project is probably going to be the only way I can fix this.

**Comment by diegomontoya** [comment]
@EricKuck Still not resolved in 1.2.2 but I will get a sample project for you to work with within the next 24 hours.

**Comment by EricKuck** [comment]
I think I might have figured it out. Is `Item` an abstract class? If so, I'm working on a fix for that right now.

**Comment by diegomontoya** [comment]
@EricKuck  

Item is not an abstract class. Just a regular class with lots of JsonFields of String/long simple types. The only thing of note is that "Item" class does contain one complex String[] JsonField.

**Comment by EricKuck** [comment]
@diegomontoya Finally did end up getting to the root cause of it, which was a race condition. This should now be fixed in 1.2.3, which was just uploaded to JCenter a second ago (so it might take a bit to become available).
